### DISCUSSION TITLE
Firefox console warning: "Cookie rejected because it is already expired."

### DISCUSSION THREAD

**Discussion by Toryn02** [gladly-team/next-firebase-auth#242] [Q&A]
Hi everyone,

I'm relatively new to implementing next-firebase-auth, and I appear to have everything working properly from a core functionality perspective (the AuthUser object is correctly being populated with Firebase Auth user data; cookies are correctly being set and unset; redirects on authed/unauthed; protecting routes server-side; etc.).  

However, when no Firebase user is currently logged into my app, I'm getting the following warnings in my Firefox browser console every time that I navigate to a page that's using withAuthUser() and useAuthUser():

```
Cookie "MyApp.AuthUserTokens" has been rejected because it is already expired.
Cookie "MyApp.AuthUserTokens.sig" has been rejected because it is already expired.
Cookie "MyApp.AuthUser" has been rejected because it is already expired.
Cookie "MyApp.AuthUser.sig" has been rejected because it is already expired.
```

The console warnings say that they are being generated from my **./pages/api/logout** file (code below):

```
import { unsetAuthCookies } from 'next-firebase-auth'
import initNextFirebaseAuth from 'lib/config/initNextFirebaseAuth'

initNextFirebaseAuth()

const handler = async (req, res) => {
  if (req.method !== 'POST') {
    return res.status(500).json({ message: 'Invalid request' })
  }

  try {
    await unsetAuthCookies(req, res)
  } catch (error) {
    return res.status(500).json({ error: error })
  }
  return res.status(200).json({ success: true })
}

export default handler
```

Example code below from one such page that is using withAuthUser() and useAuthUser() and generating the above console warnings when I navigate to it without a Firebase Auth user logged in:

```
import { useAuthUser, withAuthUser } from 'next-firebase-auth'

const Subscribe = () => {
  const AuthUser = useAuthUser()

  function handleSubmit() {
    console.log(AuthUser)
  }

  return (
    <>
      <h1>Subscribe</h1>
      <button onClick={handleSubmit}>Subscribe</button>
    </>
  )
}

export default withAuthUser()(Subscribe)
```

When I've successfully logged in using one of the Firebase Auth methods, the above warnings go away in Firefox when I visit the same page.  

I tried testing the same behavior in Chrome and the warnings do not appear there, so the issue appears to be Firefox-related.  If it's relevant, the browser that I'm using is Firefox 90.0.2 on Mac OS 10.14.6.

I'd very much appreciate any advice that you can provide on how I can resolve these Firefox console warnings.  

Thanks very much,
Josh

**Comment by kmjennison** [comment]
Hi Josh, it seems this message is just a little confusing but not an indicator of a problem. The cookies are unset successfully. See:
https://bugzilla.mozilla.org/show_bug.cgi?id=1676651
https://stackoverflow.com/q/65038173
